### ISSUE TITLE
Few bugs to fix

### ISSUE THREAD

**Issue by thosou** [pymeasure/pymeasure#170]
Hi,

I have gathered a few bugs:

* In  pymeasure/pymeasure/log.py, I think there is a mistake.
      
https://github.com/ralph-group/pymeasure/blob/e2a3c7ecae3e2c533df32e59490fb553a786e616/pymeasure/log.py#L102

It should be something like:
```
if filename is not None:
````

* If we call the status property it will return the property object. Properties cannot be added dynamically to the instance but should be added to the class:

https://github.com/ralph-group/pymeasure/blob/e2a3c7ecae3e2c533df32e59490fb553a786e616/pymeasure/instruments/instrument.py#L67-L72

Changing self to Instrument will fix this:

```
        if includeSCPI:
            # Basic SCPI commands
            Instrument.status = self.measurement("*STB?",
                                           """ Returns the status of the instrument """)
            Instrument.complete = self.measurement("*OPC?",
                                             """ TODO: Add this doc """)
````

* write_termination is missing as a safe keyword and term_chars is not supported anymore.

https://github.com/ralph-group/pymeasure/blob/e2a3c7ecae3e2c533df32e59490fb553a786e616/pymeasure/adapters/visa.py#L56-L58

* Finally I have found a typo in the IV curve tutorial:
https://pymeasure.readthedocs.io/en/latest/tutorial/procedure.html#modifying-our-script

The unit in parameter should be "units". When I copy paste this code I get an error.

I don't know how you guys want to proceed but maybe I can make a or several pull requests. I could also add more things to the SCPI check for instance.

**Comment by cjermain** [comment]
@thosou, thanks for calling out these bugs! Please make PRs as you see fit, and then request a review.

The second issue you mention with the properties being set on `self` is a good catch. The challenge here is that the Instrument class is shared among a number of other instruments, so I don't think we want to set it at that level. I would suggest using `__new__(cls, ...)` to set the properties.

**Comment by bilderbuchi** [comment]
I think you can create a custom class on the fly, see e.g. [here](https://stackoverflow.com/a/2954373/599884) or [this question](https://stackoverflow.com/questions/48448074/adding-a-property-to-an-existing-object-instance) (by @vascotenner I heavily suspect).

**Comment by vascotenner** [comment]
Properties can be dynamically added to an instance. For an example see the stackoverflow questions above, or this camera driver:
https://github.com/vascotenner/lantz/blob/drivers_developv/lantz/drivers/basler/basler.py

**Comment by thosou** [comment]
Thanks for the tip! It was good learning. So I manage to fix the problem like this:

```
        if self.SCPI is True:
            class_name = 'SCPI' + self.__class__.__name__
            child_class = type(class_name, (self.__class__,),
                               {'status': self.measurement("*STB?", """ Returns the status of the instrument """),
                                'complete': self.measurement("*OPC?", """ TODO: Add this doc """)})
            self.__class__ = child_class
```

The Instrument class is then renamed SCPIInstrument.

**Comment by bilderbuchi** [comment]
@thosou I have fixed the other issues you mentioned, so only the SCPI property issue is left. I think the approach you mention last is valid and probably the best we can do. 
I think but I'm not 100% sure if we can exclude any impact down the line when we dynamically inject a child class into the inheritance hierarchy like that -- will that work correctly with instruments inheriting from `Instrument`? @cjermain what do you think? Should we go ahead with the proposed fix?

**Comment by thosou** [comment]
Hi @bilderbuchi,
Thanks for getting back on this and sorry I put that one on hold. I think there was a small issue I was not happy with in my latest proposal. I cannot remember what but I am actually getting access to a lab soon. I will be happy to help on this.
Among the options, one could be to make 2 different classes, for SCPI instruments and none-SCPI instruments.

**Comment by bilderbuchi** [comment]
The issue with SCPI-specific properties has been fixed in #453, so we can finally close this. :partying_face:
